usecases/schema/manager.go

```diff
@@ -359,6 +360,58 @@ func (m *Manager) TenantsShards(class string, tenants ...string) (map[string]str
 	return status, err
 }
 
+// OptimisticTenantStatus tries to query the local state first. It is
+// optimistic that the state has already propagated correctly. If the state is
+// unexpected, i.e. either the tenant is not found at all or the status is
+// COLD, it will double-check with the leader.
+//
+// This way we accept false positives (for HOT tenants), but guarantee that there will never be
+// false negatives (i.e. tenants labelled as COLD that the leader thinks should
+// be HOT).
+//
+// This means:
+//
+//   - If a tenant is HOT locally (true positive), we proceed normally
+//   - If a tenant is HOT locally, but should be COLD (false positive), we still
+//     proceed. This is a conscious decision to keep the happy path free from
+//     (expensive) leader lookups.
+//   - If a tenant is not found locally, we assume it was recently created, but
+//     the state hasn't propagated yet. To verify, we check with the leader.
+//   - If a tenant is found locally, but is marked as COLD, we assume it was
+//     recently turned HOT, but the state hasn't propagated yet. To verify, we
+//     check with the leader
+//
+// Overall, we keep the (very common) happy path, free from expensive
+// leader-lookups and only fall back to the leader if the local result implies
+// an unhappy path.
+func (m *Manager) OptimisticTenantStatus(class string, tenant string) (map[string]string, error) {
+	var foundTenant bool
```

Comment:
**nitpick:** `exist` , to be read later as `!exist`

Reply:
No opinion on this from my side, happy with the suggestion. 

That said, I'd love to understand the motivation behind it: Why is `exist`/`!exist` better than `foundTenant`/`!foundTenant`?